scalyr_agent/builtin_monitors/kubernetes_events_monitor.py

```diff
@@ -0,0 +1,478 @@
+                        self._sleep_but_awaken_if_stopped(self._leader_check_interval)
+                        continue
+
+                    global_log.log( scalyr_logging.DEBUG_LEVEL_0, "Leader is %s" % (str(self._current_leader)) )
+                try:
+                    # start streaming events
+                    lines = k8s.stream_events( last_event=last_event )
+
+                    json = {}
+                    for line in lines:
+                        try:
+                            json = json_lib.parse(line)
+                        except Exception, e:
+                            global_log.warning( "Error parsing event json: %s, %s, %s" % (line, str(e), traceback.format_exc() ) )
+                            continue
+
+                        obj = json.get( "object", JsonObject() )
+
+                        # check to see if the resource version we are using has expired
+                        if json.get( 'type', '' ) == 'ERROR':
+                            if obj.get( 'kind', '' ) == 'Status' and obj.get( 'reason', '' ) == 'Expired':
+                                # if so, reset to None to get all events, we'll skip over the already seen
+                                # ones later
+                                last_event = None
+                                global_log.log( scalyr_logging.DEBUG_LEVEL_1, "K8S resource expired" )
+                                continue
+
+                        # resource version hasn't expired, so update it to the most recently seen version
+                        last_event = last_resource
+
+                        metadata = obj.get( "metadata", JsonObject() )
+
+                        # skip any events with resourceVersions higher than ones we've already seen
+                        resource_version = metadata.get_int( "resourceVersion", None, none_if_missing=True )
+                        if resource_version and resource_version <= last_resource:
+                            global_log.log( scalyr_logging.DEBUG_LEVEL_2, "Skipping older resource events" )
+                            continue
+
+                        last_resource = resource_version
+                        last_event = resource_version
+
+                        # see if this event is about an object we are interested in
+                        involved = obj.get( "involvedObject", JsonObject())
+                        kind = involved.get( 'kind', None, none_if_missing=True )
+                        if kind and kind not in self.__event_object_filter:
+                            global_log.log( scalyr_logging.DEBUG_LEVEL_1, "Ignoring event due to unknown kind %s - %s" % (kind, str(metadata) ) )
+                            continue
+
+                        # get cluster and deployment information
+                        k8s_fields = "cluster-name=%s" % cluster_name
+                        if kind:
```

Comment:
If kind is None, should we log the event at all?

Reply:
Probably not.  Kind shold always be specified, so will early `continue` if None.